Export deprecated rules in JSON

ariel-anieli opened the pull request:
<!--
Thanks for your contribution. Please make sure to fill the contents of this template with the necessary information to ease and speed up the review process.

!!! PLEASE DO NOT DELETE ANY SECTION, COMMENT OR THE CONTENT OF THE TEMPLATE. !!!
-->

### Summary of the Pull Request

Hello maintainers,

This is my first ever PR to the project; your feedback is much appreciated.

I have added JSON as export format for deprecated rules.

### Changelog

<!--
** Don't remove this comment **
You need to add one line for every changed file of the PR and prefix one of the following tags:
new:	<title>
update:	<title> - <optional comment>
fix:	<title> - <optional comment>
remove:	<title> - <optional comment>
chore: for non-detection related changes (e.g. dates/titles) and changes on workflow

e.g.
new: Brute-Force Attacks on Azure Admin Account
update: Suspicious Microsoft Office Child Process - add MSPUB.EXE
fix: Malware User Agent - remove legitimate Firefox UA
chore: workflow - update checkout version
remove: Suspicious Office Execution - deprecated in favour of 8f922766-a1d3-4b57-9966-b27de37fddd2
-->

new: deprecated/deprecated.json
update: deprecated/deprecated.csv
update: tests/deprecated_rules.py
update: deprecated/README.md

### Example Log Event

<!--
Fill this in case of false positive fixes
-->

### Fixed Issues

<!--
Link the fixed issues here, in case your commit fixes issues with rules or code
-->

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)

phantinuss commented:
Hi @ariel-anieli,

I am curious about the motivation to add this? What is your use case?

ariel-anieli commented:
> Hi @ariel-anieli,
> 
> I am curious about the motivation to add this? What is your use case?

Hi @phantinuss, thanks for your feedback. My aim was to improve the developer experience.

Does this fit into the roadmap of the project? Please let me know; if not, I'll then drop the PR.

phantinuss commented:
I am just curious about how or if these files are used as I know of no feedback. In general I don't like adding complexity for no reason but you write clean and readable code and the work is already done. So ofc we will make use of it. But I cannot promise that it will always be the case.

I made two changes:
a) if modified doesn't exist, we use date over now() as the script is regularly run (I also updated the workflow accordingly)
b) I added a secondary sort key as different systems/python versions will probably sort non-deterministically for rules that use the same modified date

Thanks for contributing. If you are looking for ways to contribute to the Sigma project, maybe you want to join the [discord server](https://discord.gg/kQQBn5W2z5) and can ask/discuss for ways to do so.

ariel-anieli commented:
> I am just curious about how or if these files are used as I know of no feedback. In general I don't like adding complexity for no reason but you write clean and readable code and the work is already done. So ofc we will make use of it. But I cannot promise that it will always be the case.

Sure, I do understand that. Thanks for your feedback.
> 
> I made two changes: a) if modified doesn't exist, we use date over now() as the script is regularly run and I don't want to have diffs to show up for these rules (I also updated the workflow accordingly) b) I added a secondary sort key as different systems/python versions will probably sort non-deterministically for rules that use the same modified date

Wonderful.
> 
> Thanks for contributing. If you are looking for ways to contribute to the Sigma project, maybe you want to join the [discord server](https://discord.gg/kQQBn5W2z5) and can ask/discuss for ways to do so.

Of course, @phantinuss; I have just joined the server. See you there!
